packages/flutter_goldens/lib/skia_client.dart

```diff
@@ -582,19 +582,22 @@ class SkiaGoldClient {
   /// the image keys.
   String getTraceID(String testName) {
     final String? webRenderer = _webRendererValue;
+    // Keys must be sorted alphabetically, after grouping by letter case.
```

Comment:
Is this also relevant for `_getKeysJSON`?

Reply:
It is not, since those keys are sorted by Gold when they are ingested. It matters here because we are using the keys to generate an md5 sum to look up the image, and being out of order means we get the wrong result.

Reply:
"alphabetically, after grouping by letter case" is just "sort by Unicode scalar value" which is just the default string sort in Dart, so one way to do this would be to create the map first, then get the keys, sort them, and add the key/value pairs to a new map in the sorted order. That would avoid future bugs when someone adds a new key manually and gets the order wrong. It would cost a few CPU cycles and some memory though.

Reply:
ah, just saw the comment below. Yeah, the order here is just the default order for Dart sorting.

```dart
void main() {
  Map<String, int> test = {'a': 3, 'B': 2, 'A': 1, 'b': 4};
  print(test);
  print(test.keys.toList()..sort());
  Map<String, int> sorted = {};
  for (String key in test.keys.toList()..sort()) {
    sorted[key] = test[key]!;
  }
  print(sorted);
}
```

->

```
{a: 3, B: 2, A: 1, b: 4}
[A, B, a, b]
{A: 1, B: 2, a: 3, b: 4}
```

Reply:
This week I learned Dart does that by default! Awesome! Yeah I'll update this to sort keys so we don't have to worry about it recurring. :)